rclcpp/test/rclcpp/test_timer.cpp

```diff
@@ -191,3 +192,20 @@ TEST_F(TestTimer, test_bad_arguments) {
     rclcpp::GenericTimer<void (*)()>(unitialized_clock, 1us, []() {}, context),
     rclcpp::exceptions::RCLError);
 }
+
+TEST_F(TestTimer, callback_with_timer) {
+  rclcpp::TimerBase * timer_ptr = nullptr;
+  timer = test_node->create_wall_timer(
+    std::chrono::milliseconds(1),
+    [&timer_ptr](rclcpp::TimerBase & timer) {
+      timer_ptr = &timer;
+    });
+  auto start = std::chrono::steady_clock::now();
+  while (nullptr == timer_ptr &&
+    (std::chrono::steady_clock::now() - start) < std::chrono::milliseconds(100))
+  {
+    executor->spin_some();
+    std::this_thread::sleep_for(std::chrono::milliseconds(10));
```

Comment:
@brawner nit: you could also use `spin_once(std::chrono::milliseconds(10))` and let it sleep.

Reply:
Switched to spin_once